framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1662,6 +1663,20 @@ protected AnnotatedDeclaredType getImplicitReceiverType(ExpressionTree tree) {
                             "AnnotatedTypeFactory.getImplicitReceiver: enclosingClass()==null for element: "
                                     + element);
                 }
+                if (tree.getKind() == Kind.NEW_CLASS) {
+                    if (typeElt.getEnclosingElement() != null) {
+                        typeElt = ElementUtils.enclosingClass(typeElt.getEnclosingElement());
+                        if (typeElt == null) {
+                            // If the typeElt does not have an enclosing class, then the
```

Comment:
The comment here and below is the same, which was a bit surprising. Would the logic be easier to understand as:

```java
if (typeElt.getEnclosingElement() == null || ElementUtils.enclosingClass(typeElt.getEnclosingElement() == null) {
    // If the typeElt does not have an enclosing class, then the NewClassTree
    // does not have an implicit receiver.
    return null;
}
```

That doesn't perform the re-assignment of `typeElt` when it has an enclosing class, which is tricky to notice.
But is that re-assignment good, as `typeElt` is used in `getEnclosingType` below?
So I guess I'm asking for some more documentation what the cases are for.